wrapper/src/main/java/software/amazon/jdbc/plugin/efm/HostMonitorImpl.java

```diff
@@ -178,8 +179,8 @@ public void run() {
           }
 
           if (!this.activeContexts.isEmpty()
-              || this.monitoringConn == null
-              || this.monitoringConn.isClosed()) {
+              || this.monitoringConn.get() == null
+              || this.monitoringConn.get().isClosed()) {
```

Comment:
I know the odds of this are low but I think that monitoringConn could become null between the first and second line, which would produce an NPE, should we introduce a local variable above to store the result of monitoringConn.get()?